### PR TITLE
Sonar upgrade to 8.9 LTS

### LINKED ISSUE #986: Simplify SonarQube config
Instead of forcing auth via API in the configure script (https://github.com/opendevstack/ods-core/blob/master/sonarqube/configure.sh#L199-L205) we can do this in the container startup script: https://github.com/opendevstack/ods-core/blob/master/sonarqube/docker/run.sh.

FYI @felipecruz91

### LINKED ISSUE #1075: Update SonarQube to latest LTS
ODS is officially compatible with SQ 8.2. It is known not to work with 8.5+ (see #960). The latest LTS is 8.9. From 8.5 onwards, the language plugins are bundled, which would reduce the maintenance effort on our side, and would fix e.g. #1073.

I think ODS should be compatible with the latest LTS whenever possible.

BTW: The latest version is 9.2, which has support for Terraform files, which sounds like a nice addition as well.

### PR DESCRIPTION
**Pull request by braisvq1996**
closes #1075
closes #986 

@gerardcl @michaelsauter @metmajer

### CODE REVIEW

File: sonarqube/docker/Dockerfile
Comment (michaelsauter) [inline sonarqube/docker/Dockerfile:74]: I assume the removal of apex is intentional? If so, maybe we can add a statement in the changelog about its removal?

Comment (gerardcl) [reply sonarqube/docker/Dockerfile:74]: right, see https://update.sonarsource.org/plugins/compatibility-matrix.html

Comment (braisvq1996) [reply sonarqube/docker/Dockerfile:74]: Some of the plugins were removed because they are now bundled in SonarQube

Comment (michaelsauter) [reply sonarqube/docker/Dockerfile:74]: ah that one is also bundled. nice!

**Review by michaelsauter — APPROVED** [review]

**Review by michaelsauter — DISMISSED** [review]

**Review by stitakis — COMMENTED** [review]
I'm disapproving this change to avoid merging it right now. The ods-core build is broken since days. I'd like to get this fixed before adding more functionality. @braisvq1996 if you have time  it would be awesome if you could have a quick look and figure out why the current build is failing. Please let me know. Thanks!

**Comment by gerardcl** [comment]
Hi @stitakis , not sure why you would block this PR, this is not a new functionality but a required upgrade, the ods-devenv issue is unrelated to this upgrade IMHO, so I don't see a need to block this based on your last comment. Sorry.

**Comment by gerardcl** [comment]
We have agreed to release this PR via ODS 4.2.0 release, so the focus will be to first fix master along with releasing current planned 4.1.0 ODS version.

**Comment by stitakis** [comment]
@gerardcl @braisvq1996 @michaelsauter to get everyone on same page... do we agree to first fix the ODS AMI build for `master` and `4.x` and then merge this PR? Is this the plan?

File: sonarqube/docker/Dockerfile
Comment (tbugfinder) [reply sonarqube/docker/Dockerfile:56]: sonar-crowd repo was archived.
So drop this software?

Comment (braisvq1996) [reply sonarqube/docker/Dockerfile:56]: The last suported version is 8.9 that is the one this PR will update to.
But you have reason that for the next LTS version (9.9) this plugin will not be available.

Comment (tbugfinder) [reply sonarqube/docker/Dockerfile:56]: So, could this PR then merged or what is left after fixing merge conflicts?

Comment (braisvq1996) [reply sonarqube/docker/Dockerfile:56]: I know this has been pending for near a year....
Right now I am trying to make this part of the next release bundle and as soon as this is done we will review the PR and merge it

**Review by gerardcl — APPROVED** [review]
LGTM! 🚀

**Comment by michaelsauter** [comment]
🎉

**Pull request merged**
